packages/core/src/Entity.ts

```
@@ -193,11 +224,17 @@ export class Entity extends EngineObject {
    * Create a entity.
    * @param engine - The engine the entity belongs to
    */
-  constructor(engine: Engine, name?: string) {
+  constructor(engine: Engine, name?: string, ...components: ComponentConstructor[]) {
     super(engine);
     this.name = name;
-    this.transform = this.addComponent(Transform);
-    this._inverseWorldMatFlag = this.transform.registerWorldChangeFlag();
+    for (let i = 0, n = components.length; i < n; i++) {
+      const type = components[i];
+      if (!Transform.prototype.isPrototypeOf(type.prototype) || !this.transform) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'isPrototypeOf' from target object.



(lint/suspicious/noPrototypeBuiltins)